third-party/realdds/include/realdds/dds-adapter-watcher.h

```diff
@@ -0,0 +1,40 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
```

Comment:
Adapter is confusing, I mixed it with rs-dds-adapter.

How about `net-adapter-watcher`?

Reply:
I can go as far as putting the rsutils class inside `rsutils::network::adapter_watcher`, maybe.
But the DDS class has nothing to do with implementations on top.
I don't want to abbreviate...
`dds_network_adapter_watcher`?

Reply:
Renamed `rsutils::os::adapter_watcher` -> `network_adapter_watcher`
Renamed `realdds::dds_adapter_watcher` -> `dds_network_adapter_watcher`

Reply:
Great